src/engine/srv_internal.h

```diff
@@ -339,4 +339,14 @@ dss_xs2tgt(int xs_id)
 	       (dss_tgt_offload_xs_nr / dss_tgt_nr + 1);
 }
 
+/*
+ * FIXME: control plane shall pass tier information to
+ * avoid create nvme context if Metadata on SSD is not enabled.
+ */
+static inline bool
+dss_xstream_has_nvme(struct dss_xstream *dx)
+{
+	return dx->dx_main_xs != 0 || dx->dx_xs_id == 0;
```

Comment:
So the per-xstream NVMe context will be created on sys xstream after this PR, I'm not sure if it could cause any test failures, since sys xstream will consume some reserved hugepages too.

It reminds me that the hugepage nr auto calculation on control plane needs be adjusted for meta data on SSD.

Reply:
Indeed